### PR TITLE
reading jar and collecting entities

### PR DESCRIPTION
**Pull request by Adrien-COSTANDI**
JarReader works and has tests associated

### CODE REVIEW

File: src/main/java/com/github/magickoders/Hello.java [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
+            var entities = JarReader.getEntities();
+            try (var writer = new OutputStreamWriter(System.out, StandardCharsets.UTF_8)) {
+                var text = entities.stream()
+                        .filter(entity -> entity.name().contains("magickoders") /*|| entity.name().contains("forax")*/)
+                        .map(Entity::toString)
+                        .collect(Collectors.joining("\n"));
+
+                writer.append(text);
+            }
+        } catch (IOException e) {
```
Comment (forax) [inline src/main/java/com/github/magickoders/Hello.java:24]: use throws instead

File: src/main/java/com/github/magickoders/JarReader.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+ * It contains a static method {@link #getEntities()} which returns the entities contained in the directory "target"
+ */
+public class JarReader {
+
+    /**
+     * A class that contains the list of entities by using the ClassVisitor's methods
+     */
+    private static class MyVisitor extends ClassVisitor { // bwaaa de l'héritage, mais c'est une class abstraite donc ok j'imagine
+
+        private final List<Entity> entities = new ArrayList<>();
```
Comment (forax) [inline src/main/java/com/github/magickoders/JarReader.java:30]: should be declared as an ArrayList not a List, the field is private, not visible from outside

File: src/main/java/com/github/magickoders/JarReader.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+
+        public List<Entity> getEntities() {
+            return entities;
+        }
+
+        protected MyVisitor(int api) {
```
Comment (forax) [inline src/main/java/com/github/magickoders/JarReader.java:37]: protected is useless, given that the class is private, the constructor should be private

File: src/main/java/com/github/magickoders/JarReader.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        /**
+         * method that returns the modifiers from the access obtained by a visit method of {@link MyVisitor}
+         *
+         * @param access the value obtained by {@link #visit(int, int, String, String, String, String[])}
+         * @return a set of modifiers which can contain private, static, public...
+         * @see Modifier
+         */
+        private static Set<Modifier> getModifiers(int access) {
+            var modifiers = new HashSet<Modifier>();
+            if (java.lang.reflect.Modifier.isPublic(access)) {
```
Comment (forax) [inline src/main/java/com/github/magickoders/JarReader.java:69]: isolate that code in a private method

File: src/main/java/com/github/magickoders/JarReader.java [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        }
+
+        /**
+         * method that returns the specific stereotype of a class. If it is a simple class, returns Optional.empty()
+         *
+         * @param access the value obtained by {@link #visit(int, int, String, String, String, String[])}
+         * @return an Optional which contains the stereotype of the class. Optional.empty() if it is a simple class
+         */
+        private static Optional<String> getStereotype(int access) {
+            if ((access & Opcodes.ACC_RECORD) != 0) {
```
Comment (forax) [inline src/main/java/com/github/magickoders/JarReader.java:93]: isolate that code in a private method

File: src/main/java/com/github/magickoders/JarReader.java [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+    /**
+     * This method returns the list of entities in the directory "target"
+     *
+     * @return the list of entities contained in the directory "target"
+     * @throws IOException
+     *
+     * @see #getEntities(String)
+     */
+    public static List<Entity> getEntities() throws IOException {
+        return getEntities("target");
```
Comment (forax) [inline src/main/java/com/github/magickoders/JarReader.java:116]: using the string "target" is ugly, all configuration should be taken as parameter of the constructor or the method

File: src/main/java/com/github/magickoders/JarReader.java [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+        var finder = ModuleFinder.of(path);
+        var myVisitor = new MyVisitor(Opcodes.ASM9);
+
+        for(var moduleReference: finder.findAll()) {
+            try(var reader = moduleReference.open()) {
+                for(var filename: (Iterable<String>) reader.list()::iterator) {
+                    if (!filename.endsWith(".class")) {
+                        continue;
+                    }
+                    // ouvre la classe
```
Comment (forax) [inline src/main/java/com/github/magickoders/JarReader.java:140]: please use English

**Comment by forax** [comment]
I will accept your PR once you have fixed the code

**Comment by forax** [comment]
You also need to fix all these warnings from CheckStyle

[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:9:1: Extra separation in import group before 'java.io.IOException' [CustomImportOrder]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:9:1: Wrong lexicographical order for 'java.io.IOException' import. Should be before 'org.objectweb.asm.Opcodes'. [CustomImportOrder]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:10:1: Wrong lexicographical order for 'java.lang.module.ModuleFinder' import. Should be before 'org.objectweb.asm.Opcodes'. [CustomImportOrder]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:11:1: Wrong lexicographical order for 'java.nio.file.Path' import. Should be before 'org.objectweb.asm.Opcodes'. [CustomImportOrder]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:12:1: Wrong lexicographical order for 'java.util.ArrayList' import. Should be before 'org.objectweb.asm.Opcodes'. [CustomImportOrder]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:13:1: Wrong lexicographical order for 'java.util.HashSet' import. Should be before 'org.objectweb.asm.Opcodes'. [CustomImportOrder]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:14:1: Wrong lexicographical order for 'java.util.List' import. Should be before 'org.objectweb.asm.Opcodes'. [CustomImportOrder]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:15:1: Wrong lexicographical order for 'java.util.Objects' import. Should be before 'org.objectweb.asm.Opcodes'. [CustomImportOrder]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:16:1: Wrong lexicographical order for 'java.util.Optional' import. Should be before 'org.objectweb.asm.Opcodes'. [CustomImportOrder]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:17:1: Wrong lexicographical order for 'java.util.Set' import. Should be before 'org.objectweb.asm.Opcodes'. [CustomImportOrder]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:21: Line is longer than 100 characters (found 116). [LineLength]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:25: First sentence of Javadoc is missing an ending period. [SummaryJavadoc]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:28: Line is longer than 100 characters (found 130). [LineLength]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:28:5: 'class def modifier' has incorrect indentation level 4, expected level should be 2. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:30:9: 'member def modifier' has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:33:9: 'method def modifier' has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:34:13: 'method def' child has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:35:9: 'method def rcurly' has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:37:9: 'ctor def modifier' has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:38:13: 'ctor def' child has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:39:9: 'ctor def rcurly' has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:53:9: 'method def modifier' has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:54: Line is longer than 100 characters (found 122). [LineLength]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:55: Line is longer than 100 characters (found 127). [LineLength]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:55:13: 'method def' child has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:56:13: 'method def' child has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:57:9: 'method def rcurly' has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:60: First sentence of Javadoc is missing an ending period. [SummaryJavadoc]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:61: Line is longer than 100 characters (found 108). [LineLength]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:63: Line is longer than 100 characters (found 105). [LineLength]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:67:9: 'method def modifier' has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:68:13: 'method def' child has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:69:13: 'if' has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:70:17: 'if' child has incorrect indentation level 16, expected level should be 8. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:71:13: 'if rcurly' has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:72:17: 'if' child has incorrect indentation level 16, expected level should be 8. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:73:13: 'if rcurly' has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:74:17: 'if' child has incorrect indentation level 16, expected level should be 8. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:75:13: 'if rcurly' has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:76:17: 'else' child has incorrect indentation level 16, expected level should be 8. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:77:13: 'else rcurly' has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:79:13: 'if' has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:80:17: 'if' child has incorrect indentation level 16, expected level should be 8. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:81:13: 'if rcurly' has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:83:13: 'method def' child has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:84:9: 'method def rcurly' has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:87: Line is longer than 100 characters (found 116). [LineLength]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:89: Line is longer than 100 characters (found 105). [LineLength]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:90: Line is longer than 100 characters (found 115). [LineLength]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:92:9: 'method def modifier' has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:93:13: 'if' has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:94:17: 'if' child has incorrect indentation level 16, expected level should be 8. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:95:13: 'if rcurly' has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:96:17: 'if' child has incorrect indentation level 16, expected level should be 8. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:97:13: 'if rcurly' has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:98:17: 'if' child has incorrect indentation level 16, expected level should be 8. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:99:13: 'if rcurly' has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:100:17: 'if' child has incorrect indentation level 16, expected level should be 8. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:101:13: 'if rcurly' has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:102:17: 'else' child has incorrect indentation level 16, expected level should be 8. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:103:13: 'else rcurly' has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:104:9: 'method def rcurly' has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:105:5: 'class def rcurly' has incorrect indentation level 4, expected level should be 2. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:107: First sentence of Javadoc is missing an ending period. [SummaryJavadoc]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:115:5: 'method def modifier' has incorrect indentation level 4, expected level should be 2. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:116:9: 'method def' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:117:5: 'method def rcurly' has incorrect indentation level 4, expected level should be 2. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:120: First sentence of Javadoc is missing an ending period. [SummaryJavadoc]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:126: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:128:5: 'method def modifier' has incorrect indentation level 4, expected level should be 2. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:129:9: 'method def' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:130:9: 'method def' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:131:9: 'method def' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:132:9: 'method def' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:134:9: 'for' has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:134:9: 'for' is not followed by whitespace. [WhitespaceAfter]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:134:9: WhitespaceAround: 'for' is not followed by whitespace. Empty blocks                may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:134:32: WhitespaceAround: ':' is not preceded with whitespace. [WhitespaceAround]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:135:13: 'try' has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:135:13: 'try' is not followed by whitespace. [WhitespaceAfter]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:135:13: WhitespaceAround: 'try' is not followed by whitespace. Empty blocks                may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:136:17: 'for' has incorrect indentation level 16, expected level should be 8. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:136:17: 'for' is not followed by whitespace. [WhitespaceAfter]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:136:17: WhitespaceAround: 'for' is not followed by whitespace. Empty blocks                may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:136:33: WhitespaceAround: ':' is not preceded with whitespace. [WhitespaceAround]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:137:21: 'if' has incorrect indentation level 20, expected level should be 10. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:138:25: 'if' child has incorrect indentation level 24, expected level should be 12. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:139:21: 'if rcurly' has incorrect indentation level 20, expected level should be 10. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:141:21: 'try' has incorrect indentation level 20, expected level should be 10. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:141:21: 'try' is not followed by whitespace. [WhitespaceAfter]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:141:21: WhitespaceAround: 'try' is not followed by whitespace. Empty blocks                may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:142:25: 'try' child has incorrect indentation level 24, expected level should be 12. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:143:25: 'try' child has incorrect indentation level 24, expected level should be 12. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:144:21: 'try rcurly' has incorrect indentation level 20, expected level should be 10. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:145:17: 'for rcurly' has incorrect indentation level 16, expected level should be 8. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:146:13: 'try rcurly' has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:147:9: 'for rcurly' has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:149:9: 'method def' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/JarReader.java:150:5: 'method def rcurly' has incorrect indentation level 4, expected level should be 2. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/Hello.java:5:1: Extra separation in import group before 'java.io.IOException' [CustomImportOrder]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/Hello.java:10:1: Missing a Javadoc comment. [MissingJavadocType]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/Hello.java:11:5: 'method def modifier' has incorrect indentation level 4, expected level should be 2. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/Hello.java:11:5: Missing a Javadoc comment. [MissingJavadocMethod]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/Hello.java:12:9: 'method def' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/Hello.java:14:9: 'try' has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/Hello.java:15:13: 'try' child has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/Hello.java:16:13: 'try' has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/Hello.java:17:17: 'try' child has incorrect indentation level 16, expected level should be 8. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/Hello.java:18: Line is longer than 100 characters (found 119). [LineLength]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/Hello.java:22:17: 'try' child has incorrect indentation level 16, expected level should be 8. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/Hello.java:23:13: 'try rcurly' has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/Hello.java:24:9: 'try rcurly' has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/Hello.java:25:13: 'catch' child has incorrect indentation level 12, expected level should be 6. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/Hello.java:26:9: 'catch rcurly' has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/runner/work/umldoc/umldoc/src/main/java/com/github/magickoders/Hello.java:27:5: 'method def rcurly' has incorrect indentation level 4, expected level should be 2. [Indentation]

**Comment by Adrien-COSTANDI** [comment]
I have fixed the errors you pointed and the checkstyle warnings

**Pull request merged**